src/coreclr/jit/gentree.h

```diff
@@ -487,6 +487,8 @@ enum GenTreeFlags : unsigned int
                                               //           This flag is useful in cases where it is required to generate register
                                               //           indirect addressing mode. One such case is virtual stub calls on xarch.
     GTF_IND_UNALIGNED           = 0x02000000, // OperIsIndir() -- the load or store is unaligned (we assume worst case alignment of 1 byte)
+    GTF_IND_KNOWN_UNALIGNED     = 0x00100000, // GT_IND -- same as GTF_IND_UNALIGNED (it's expected to be set) with a legal allowance
+                                              //           to use wider loads/stores where needed.
```

Comment:
`IND_KNOWN_UNALIGNED` is a pretty confusing name in my opinion. We don't actually know if it's unaligned dynamically.
```suggestion
    GTF_IND_ALLOW_NON_ATOMIC    = 0x00100000, // GT_IND -- this memory access does not need be atomic
```
Nit: the flags are sorted by value. It would be good to maintain that.
Nit: you should add this to `GTF_IND_FLAGS`.

Reply:
Makes sense! Will apply once I get the diffs, want to know how bad they're 😄

Reply:
Ah, the diffs are not too bad https://dev.azure.com/dnceng-public/public/_build/results?buildId=426134&view=ms.vss-build-web.run-extensions-tab, initial improvements were +400kb per platform.